### DISCUSSION TITLE
Update config upon component state change

### DISCUSSION THREAD

**Discussion by ReeceHumphreys** [markdoc/markdoc#314] [Q&A] [answered]
I am currently trying to do conditional content displaying depending on what version of the API the user has indicated. The user selects `1.0` or `2.0` in a side navigation component that uses a ReactContext to keep track of the state. Additionally, this state gets stored in localstorage.

It seems that config variables are what I am looking for, so I can do something along the lines of:

```
{% if $flags.version2 %}
...
{% /if %}

{% if $flags.version1 %}
...
{% /if %}
```

However, I need help finding the optimal way to update the config variables on state change. I created a `config.js` but cannot use the Context since the config is not a React Component. I am using nextjs, and the application architecture is very similar to the `nextjs-starter` that Markdoc provides. Any guidance would be much appreciated!

**Comment by rpaul-stripe** [comment, marked as answer]
Markdoc conditionals and variables are typically resolved on the server side rather than the client side, so they can't take advantage of client state. For things that need to update on client-side state changes, you typically want to make a React component and then wrap it with a custom tag.

**Reply by ReeceHumphreys** [reply]
Okay thank you for the answer! If I was to resolve the variables on the server side I am still a bit unclear as to how I would update the configuration to include these variables in the `config.js` file.

**Reply by mfix-stripe** [reply]
If you have a `markdoc/config.js`, you can export a `const variables = {...}` object from that file to include those variables. 

If you have other dynamic state, I would encourage you to manage that using general React/Next.js state patterns, in `useState`, context, potentially storing it in `_app.js` if you need the state to persist between page changes.
